htdocs/core/class/stats.class.php

```diff
@@ -181,6 +188,10 @@ public function getAmountByMonthWithPrevYear($endyear, $startyear, $cachedelay =
         else
 		{
 			$year=$startyear;
+			if (empty($conf->global->GRAPH_USE_FISCAL_YEAR)) {
+				$startmonth = 0;
+			}
+			if ($startmonth != 0) $year = $year - 1;
```

Comment:
If $startmonth is 1 (january), we should have same resutl than if $conf->global->GRAPH_USE_FISCAL_YEAR is not set.
But here we have a difference of 1 year due to next line.
Can you check ?

Reply:
Hi Eldy
No, if January, $startmonth is 0.
See https://github.com/Dolibarr/dolibarr/blob/053ae8224b45b0940d0539e35a4b9be5bfacfab3/htdocs/core/boxes/box_graph_invoices_permonth.php#L77
		$startmonth = $conf->global->SOCIETE_FISCAL_MONTH_START?($conf->global->SOCIETE_FISCAL_MONTH_START-1):0;
Alexandre modified my comments about variable, but $startmonth is from 0 to 11.

Reply:
In php January is 1

Reply:
@eldy @hregis Propose a new code with min 1 max 12 with 1 = january